### DISCUSSION TITLE
navigatorKey.currentState.push equivalent

### DISCUSSION THREAD

**Discussion by RaduGrama** [csells/go_router#258] [Q&A]
Is there a `navigatorKey.currentState.push()` equivalent in go_router for scenarios where navigation is triggered from Redux middleware for example? With `navigatorKey` defined as `final navigatorKey = GlobalKey<NavigatorState>();`

**Comment by csells** [comment]
what problem are you trying to solve?

**Comment by RaduGrama** [comment]
I use Redux for state management and I have epics that handle certain Redux actions, where a context is not available. When I initialize each epic, I pass in the global `navigatorKey` and then from epics I can call `navigatorKey.currentState.push()` to push a new screen.

I think I answered my own question by using code like:

```
final router = GoRouter(routes: [ /* routes */ ]);
final navigatorKey = router.routerDelegate.navigatorKey;

// original code
```

**Comment by csells** [comment]
And did that do the trick?
